src/benchmark.cc

```diff
@@ -110,6 +140,37 @@ namespace benchmark {
 
 namespace internal {
 
+// Performance measurements always come with random variances. Defines a
+// factor by which the required number of iterations is overestimated in order
+// to reduce the probability that the minimum time requirement will not be met.
+const double kSafetyMultiplier = 1.4;
+
+// Wraps --benchmark_min_time and returns valid default values if not supplied.
+double GetMinTime() {
+  const double min_time = FLAGS_benchmark_min_time;
+  if (min_time >= 0.0) {
+    return min_time;
+  }
+
+  if (FLAGS_benchmark_enable_random_interleaving) {
```

Comment:
ah so if i enable random interleaving i can't override the minimum time? that doesn't seem right. i thought this would be:

```c++
const double min_time = (FLAGS_benchmark_min_time >= 0.0 ? min_time : kDefaultMinTimeTotal);
return min_time / GetRepetitions();
```

Reply:
This is handled in the if statement above,

1. min_time = FLAG_min_time
2.  if min_time >= 0.0, return min_time, so no matter random_interleaving is set or not, I return flag_min_time if it is >= 0.0

Reply:
ah i see. so the user selection of min time completely overrides anything that random interleaving would try to do. ie, the default is 0.5, but with interleaving on it's actually 0.5/12. the user sets it to 1.0, and it's then actually 1.0 (not 1.0 / 12).

i think this makes sense but it may be a little confusing (it was for me). any idea how we could make this clearer (if you agree it needs to be)?

Reply:
How about now?